libbpf-rs/src/map.rs

```diff
@@ -194,20 +196,34 @@ impl Map {
             return Err(Error::System(-fd));
         }
 
-        let ty = unsafe { libbpf_sys::bpf_map__type(ptr.as_ptr()) };
-        let key_size = unsafe { libbpf_sys::bpf_map__key_size(ptr.as_ptr()) };
-        let value_size = unsafe { libbpf_sys::bpf_map__value_size(ptr.as_ptr()) };
+        let info = Self::info(fd)?;
 
         Ok(Map {
             fd,
             name,
-            ty,
-            key_size,
-            value_size,
+            ty: info.type_ as _,
+            key_size: info.key_size,
+            value_size: info.value_size,
+            max_entries: info.max_entries,
             ptr: Some(ptr),
         })
     }
 
+    fn info(fd: i32) -> Result<libbpf_sys::bpf_map_info> {
```

Comment:
We just recently added the `MapInfo` type and it should be usable here instead, I would think.

https://github.com/libbpf/libbpf-rs/blob/30ae9cd0c8f5ba643f219432d9fb7778cc2db0e8/libbpf-rs/src/map.rs#L744-L767